lib/EarlyFunctionAnalysis/DetectABI.cpp

```diff
@@ -686,6 +689,8 @@ void DetectABI::finalizeModel() {
     auto NewType = makeType<RawFunctionType>();
     auto &FunctionType = *llvm::cast<RawFunctionType>(NewType.get());
 
+    FunctionType.Architecture() = Arch;
```

Comment:
You can infer the architecture from the `MetaAddress` of `model::Function::Entry()`.

Reply:
Done.